internal/checker/cloudformation/rules/networking/verify_no_public_ip_in_ec2_instance.rego

```diff
@@ -0,0 +1,58 @@
+# METADATA
+# title: "Verify no public IP in EC2 instance."
+# description: "Confirm that an EC2 instance has no public IP address assigned for enhanced security."
+# related_resources:
+# - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-instance.html
+# - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-launchtemplate.html
+# custom:
+#   ID: CB_CFT_069
+#   Severity: HIGH
+package lib.cloudformation.CB_CFT_069
+import future.keywords.in
+
+resource := ["AWS::EC2::Instance", "AWS::EC2::LaunchTemplate"]
+
+is_valid {
+	some resources in input.Resources
+	resources.Type in resource
+}
+
+fail[instance] {
+	is_valid
+	some resources in input.Resources
+    resources.Type in resource
+    instance := resources.Properties
+    some assoc_pub in instance.NetworkInterfaces
+    assoc_pub.AssociatePublicIpAddress == true
+}
+
```

Comment:
do this for multiple resource support
is_valid {
	supported_resources := ["AWS::EC2::Instance", "AWS::EC2::LaunchTemplate"]
	some resources in input.Resources
	resources.Type in supported_resources
}

resource[type] {
	some resource in input.Resources
    is_valid
    count(fail) > 0
    type := resource.Type
}

resource[type] {
	some resource in input.Resources
    is_valid
    count(pass) > 0
    type := resource.Type
}